pkg/compare/compare_test.go

```diff
@@ -149,6 +149,7 @@ type Test struct {
 	shouldDiffAll         bool
 	outputFormat          string
 	checks                Checks
+	showMore              bool
```

Comment:
instead of a flag lets move it to be a field in the metadata.yaml
because the reference is built in a different way depending on the flag
also least include the option to disable/enable this option for separate CRs,  we can also do it for Components but as it looks to me in the example we have sometimes only on CR in  a component is more sensitive needing for an exact match.


I have another proposal to consider.:

adding for each template in the reference an option to specify for which sections to do the merge, here is an example:

```
Parts:
  - name: ExamplePart
    Components:
      - name: DemonSets
        type: Required
        requiredTemplates:
          - apps.v1.DaemonSet.kube-system.kindnet.yaml:
            MergeAttributes:
                   - - metadata
                     - annotations
                    - - metadata
                     - lables            
        optionalTemplates:
          - apps.v1.DaemonSet.kube-system.kindnet2.yaml

``` 
In case we want to use it for the whole file we can use "." or something else just wanted to open this derection) 
 I think this i very handy, that way CRs with problematic sections still can use the merges for some fields.
 This is the general idea. 
 
 summary: I think it should be in the metadata.yaml instead, and include an option for more options of configuration of the feature on different CRs   and we should do another pass on the reference for are case studies again to understand the situation where we would like an exact match

Reply:
I think the flag is still a useful tool for people building references to see the effect of merging and also for people who want to expose any diffs hidden by merging.

Reply:
totally agree with you that users should not change the reference! 
its even exactly my point, this setting is part of the reference, the reference builder is fully in charge of it.
 the reference builder is in charge complete of the full testing of the reference, meaning if its set to ok to merge then it is ok.
 
 i don't agree that it is compatible to verbosity, by this approach you can take i to the direction of the extreme and say the same about any other rule in the reff, and we dont have a flag for that ( or the example i gave for the fields to ommit, its also possible users would like to view things there that were chose to be supressed by the ref builder)  

but totally i am happy to hear you opinion! continue with the great work! and yeah we can revisit everything, and try to merge in small parts the things we are sure about